### ISSUE TITLE
`env` and `vars` sections are evaluated without checking `platforms`

### ISSUE THREAD

**Issue by JamieEdge** [go-task/task#1273]
The `env` and `vars` sections of a task are always resolved, even if the the current platform is not listed in `platforms`.

This is reproducible using the following Taskfile.

```yaml
version: 3

tasks:
  default:
    - task: windows
    - task: darwin

  windows:
    internal: true
    platforms:
      - windows
    env:
      EXAMPLE_ENV:
        sh: exit 1
    vars:
      EXAMPLE_VAR:
        sh: exit 2
    cmds:
      - echo Windows

  darwin:
    internal: true
    platforms:
      - darwin
    cmds:
      - echo Darwin
```

The issue is shown in the following output.

```
$ uname
Darwin
$ task
task: Failed to run task "default": task: Command "exit 2" failed: exit status 2
```

This is problematic when setting a value to the output of a platform-specific command (e.g. the output of `mktemp`, which is not available on Windows).

- Task version: v3.27.1 (h1:cftsoOqUo7/pCdtO7fDa4HreXKDvbrRhfhhha8bH9xc=)
- Operating system: macOS 13.4.1 (22F82)
- Experiments enabled: None

**Comment by JamieEdge** [comment]
It's possible to workaround this issue by using the Go template engine (e.g. `{{if eq OS "windows"}}command{{end}}`).

**Comment by andreynering** [comment]
Hi @JamieEdge,

Thanks for reporting. Fixed by #1377.
